src/test/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseTestRepository.java

```diff
@@ -14,30 +14,36 @@
 public interface ProgrammingExerciseTestRepository extends JpaRepository<ProgrammingExercise, Long> {
 
     @Query("""
-            select p from ProgrammingExercise p
-            left join fetch p.studentParticipations
-            left join fetch p.attachments
-            left join fetch p.categories
-            left join fetch p.templateParticipation
-            left join fetch p.solutionParticipation
-            left join fetch p.exampleSubmissions
-            left join fetch p.exerciseHints eh
-            left join fetch eh.solutionEntries
-            left join fetch p.tutorParticipations
-            left join fetch p.posts
-            left join fetch p.testCases tc
-            left join fetch tc.solutionEntries
-            left join fetch p.staticCodeAnalysisCategories
-            left join fetch p.auxiliaryRepositories
-            left join fetch p.tasks t
-            left join fetch t.testCases
-            left join fetch t.exerciseHints
-            left join fetch p.plagiarismDetectionConfig
-            where p.id = :#{#exerciseId}
+            SELECT p
+            FROM ProgrammingExercise p
+                LEFT JOIN FETCH p.studentParticipations
+                LEFT JOIN FETCH p.attachments
+                LEFT JOIN FETCH p.categories
+                LEFT JOIN FETCH p.templateParticipation
+                LEFT JOIN FETCH p.solutionParticipation
+                LEFT JOIN FETCH p.exampleSubmissions
+                LEFT JOIN FETCH p.exerciseHints eh
+                LEFT JOIN FETCH eh.solutionEntries
+                LEFT JOIN FETCH p.tutorParticipations
+                LEFT JOIN FETCH p.posts
+                LEFT JOIN FETCH p.testCases tc
+                LEFT JOIN FETCH tc.solutionEntries
+                LEFT JOIN FETCH p.staticCodeAnalysisCategories
+                LEFT JOIN FETCH p.auxiliaryRepositories
+                LEFT JOIN FETCH p.tasks t
+                LEFT JOIN FETCH t.testCases
+                LEFT JOIN FETCH t.exerciseHints
+                LEFT JOIN FETCH p.plagiarismDetectionConfig
+            WHERE p.id = :exerciseId
             """)
     ProgrammingExercise findOneWithEagerEverything(@Param("exerciseId") Long exerciseId);
```

Comment:
```suggestion
    ProgrammingExercise findOneWithEagerEverything(@Param("exerciseId") long exerciseId);
```

Reply:
This is a lot of `LEFT JOIN FETCH` can't we use an entity graph here?

Reply:
I tried using an EntityGraph but this did not work since `exerciseHints.solutionEntries` is not valid (this attribute only exists on the subclass CodeHint).